### DISCUSSION TITLE
Dependency or dev dependency?

### DISCUSSION THREAD

**Discussion by maxdzin** [Baroshem/nuxt-security#440] [General]
Hi @Baroshem!
I'm glad the module continues developing and evolving.
I have a question regarding the dependencies. I can see that with the latest updates it is recommended way to install it via the command:
```bash
npx nuxi@latest module add nuxt-security
``` 
which installed the module as dependency, rather than dev dependency as it was previously. Is that correct? If so, can you please explain to me the reason for moving it from dev deps into dev.

Thank you!

**Comment by Baroshem** [comment]
Hey @maxdzin 

Glad you are still with us!

That is quite interesting, in my case, when installing the module like this it is installed as a dev depenendency 
![image](https://github.com/Baroshem/nuxt-security/assets/37120330/36dbbb5d-bcc1-4ed6-8c03-4d1fc54469a3)

In general, nuxt modules right now can be safely installed as dev dependencies because they are used to build the project and then they are not needed further. So I would keep it as a dev dependency still.

I think we may have some issue with nuxi command as other user reported that there is a problem that the module cannot be located #436

**Reply by Baroshem** [reply]
@vejja how does it look for you? Is it also installed as a dep or devDep?

**Reply by maxdzin** [reply]
@Baroshem thank you for the clarification. Yes, that's interesting. I thought the same, but recently I have such problems with other modules as well, so not only `security`. It seems related to `nuxi module add` command.
I'll check that more.

**Reply by maxdzin** [reply]
>nuxt modules right now can be safely installed as dev dependencies because they are used to build the project and then they are not needed further

That's also confusing because in some other modules (content, robots, ...) it is stated that it needs to be added as a dependency instead of dev dependency. Maybe there are some reasons for that, but I'm not sure.

**Comment by maxdzin** [comment]
@Baroshem I just saw that a corresponding task was created: https://github.com/nuxt/cli/issues/399

**Comment by vejja** [comment]
Just saw this : https://github.com/nuxt/cli/pull/362

So the rule is : if Nuxt is a devDependency, the module will be added as devDependency also.
Checked that and it works correctly.

The only annoying thing is that `nuxi init` sets Nuxt as a dependency whereas the docs say it should always be a devDependency.
See docs on `package.json` : https://nuxt.com/docs/guide/directory-structure/package
And docs on modules : https://nuxt.com/docs/guide/going-further/layers#npm-package

@danielroe : is it correct that Nuxt should always be a devDependency ?

**Reply by maxdzin** [reply]
And the same question for `vue` and `vue-router`. Do those need to be added to dev deps or deps?

**Comment by Baroshem** [comment]
@maxdzin 

I talked with Daniel and he said that there is no actual difference in terms of Nuxt modules to be dev dependency or normal dependency :)

**Reply by maxdzin** [reply]
@Baroshem thank you for clarifying 🙏 That's good to know about the modules.
It is interesting about `nuxt` dependency as well.
